workflow/controller/artifact_gc.go

```diff
@@ -154,7 +154,7 @@ func (woc *wfOperationCtx) processArtifactGCStrategy(ctx context.Context, strate
 		}
 		template, found := templatesByName[templateName]
 		if !found {
-			template = woc.execWf.GetTemplateByName(templateName)
+			template = woc.wf.GetTemplateByName(templateName)
```

Comment:
I think  `execWf.GetTemplateByName` is correct. execWf will carry the runtime workflow spec only. It will not have status. Code needs to refer woc.wf.status for `status`

Reply:
ignore my comments `GetTemplateByName` covers everything. you are good